js/deer-utils.js

```diff
@@ -167,11 +167,31 @@ export default {
      * Execute query for any annotations in RERUM which target the
      * id passed in. Promise resolves to an array of annotations.
      * @param {String} id URI for the targeted entity
+     * @param [String] targetStyle other formats of resource targeting.  May be null
      */
-    findByTargetId: async function (id) {
+    findByTargetId: async function (id, targetStyle) {
         let everything = Object.keys(localStorage).map(k => JSON.parse(localStorage.getItem(k)))
-        let obj = {
-            target: id
+        //Backwards compatible without the new parameter
+        if(targetStyle === undefined){
+            targetStyle = []
+        }
+        //Users pass the craziest things...
+        if (!Array.isArray(targetStyle)) {
+            if(typeof targetStyle === "string"){
+                targetStyle = [targetStyle]
+            }
+            else{
+                //certainly your key isn't a number.  keys arent objects.  what are you doing?
+                targetStyle = []
+            }
+        }
+        targetStyle = targetStyle.concat(["target", "target.@id", "target.id"]) //target.source?
```

Comment:
Does `target.@id` work? It doesn't require `[]` like in true JS?

Reply:
Yes.  mongo querying does not support bracket notation.  They only characters that can't be used in `key` names are `$` and `.` so that this is supported across special characters.


![fields](https://user-images.githubusercontent.com/3287006/56820953-d6906c80-6812-11e9-932e-be28c8a8bbd6.png)